File: JAVA/Factorial/src/Factorial.java
import java.util.Scanner;

public class Factorial {
    private static final Scanner sc;
    static {
        sc = new Scanner(System.in);
    }

    public static void main(String[] args) {
        long n;
        System.out.print("Enter a number to find its factorial -> ");
        n = sc.nextLong();
        System.out.printf("%d! = %d", n, factorial(n));
    }

    private static long factorial(long n) {
        long fac = 1;
        while(n > 0) {
            fac *= n;
            n--;
        }
        return fac;
    }
}

File: C programs/factorial.c
#include<stdio.h>
int main()
{
    int n, factorial=1;
    printf("Enter a number to calculate its factorial :: ");
    scanf("%d",&n);
    for ( int i = 1; i <= n; i++)
    {
       factorial = factorial * i;
    }
    printf("The factorial of the entered number = %d",factorial);
    return 0;
}
